### ISSUE TITLE
'service' __virtual__ returned False: No service execution module loaded: check support for service management on Amazon Linux AMI-2018

### ISSUE THREAD

**Issue by dkpiesak** [saltstack/salt#47561]
### Description of Issue/Question
I've got the below two states:

salt-minion.sls
```
salt-minion:
  pkg:
    - installed
  service:
    - running
    - enable: True
    - name: salt-minion
    - require:
      - file: /etc/salt/minion
      - cmd: import_saltminion_key_minion
    - watch:
      - file: /etc/salt/minion
```

ssh.sls
```
sshd:
  service.running:
    - enable: True
    - require:
      - pkg: openssh
      - pkg: openssh-clients
      - pkg: openssh-server
      - file: /etc/ssh/sshd_config
```

When running `state.highstate test=True`, both state fail with the below errors:

```
----------
          ID: salt-minion
    Function: service.running
      Result: False
     Comment: State 'service.running' was not found in SLS 'path.to.state.salt-minion'
              Reason: 'service' __virtual__ returned False: No service execution module loaded: check support for service management on Amazon Linux AMI-2018
     Changes:

----------
          ID: sshd
    Function: service.running
      Result: False
     Comment: State 'service.running' was not found in SLS 'path.to.state.ssh'
              Reason: 'service' __virtual__ returned False: No service execution module loaded: check support for service management on Amazon Linux AMI-2018
     Changes:
```

I'm running the latest version of Saltstack:
salt-master 2018.3.0 (Oxygen)
salt-minion 2018.3.0 (Oxygen)

Minion and Master are running the same version of OS: Amazon Linux AMI release 2018.03  (different EC2 instances)

I read in some other ticket that this type of error had already been solved. 
What the reason for this failing? State `service.running` is still supported in Oxygen version from what i see in the documentation.

Thanks,

**Comment by garethgreenaway** [comment]
This should be fixed in https://github.com/saltstack/salt/pull/47271

**Comment by dkpiesak** [comment]
@garethgreenaway thanks! Thanks to your comment, I've implemented temporary fix mentioned in #47258 

Adding to minion config file the below, seems to work:
```
providers:
  service: rh_service
```

**Comment by dwiest** [comment]
I just ran into this problem with the 2017.7.5 release.  Thanks for the info @dkpiesak, this issue was holding up some testing in our environment.
